src/main/java/org/folio/circulation/domain/notice/schedule/DueDateNotRealTimeScheduledNoticeHandler.java

```diff
@@ -63,14 +64,23 @@ public DueDateNotRealTimeScheduledNoticeHandler(
   }
 
   private CompletableFuture<Result<Void>> handleNoticeGroup(List<ScheduledNotice> noticeGroup) {
-  return allOf(noticeGroup, this::getContext)
+  return allResultsOf(noticeGroup, this::getContext)
+      .thenCompose(this::handleFailures)
       .thenCompose(r -> r.after(this::sendGroupedNotice))
       .thenCompose(r -> r.after(this::updateGroupedNotice))
       .thenApply(mapResult(p -> null));
   }
 
+  private CompletableFuture<Result<List<Pair<ScheduledNotice, LoanAndRelatedRecords>>>> handleFailures(
+    List<Result<Pair<ScheduledNotice, LoanAndRelatedRecords>>> results) {
+    results.removeIf(r -> dueDateScheduledNoticeHandler.failedToFindRecordOfType(r, REQUIRED_REC_TYPES));
+    return CompletableFuture.completedFuture(results)
+      .thenApply(Result::combineAll);
+  }
+
   private CompletableFuture<Result<Pair<ScheduledNotice, LoanAndRelatedRecords>>> getContext(ScheduledNotice notice) {
     return loanRepository.getById(notice.getLoanId())
+      .thenCompose(r -> dueDateScheduledNoticeHandler.deleteNoticeIfLoanIsMissingOrIncomplete(r, notice))
```

Comment:
Do we really want to delete it here?

An alternative could be to use the `handleFailures` method to partition the results by success and failure, map failure to deletion (if we are sure we want to do that, rather than mark them in some way) and success to sending.

Reply:
You are right, it does sound better. One of the priorities of this implementation was to achieve maximum code reuse between real-time and non-real-time notice handlers. I tried the approach you suggested, and while it worked just fine in the former, it has proven to be more verbose and complicated in the latter. For instance, when we get to the method `handleFailures`, we have no idea which notice a failed result result belongs to. Yes, we can work around this, but it just makes things overly complicated.
In short, even though the approach I used is not the most straightforward, deleting notices right on the spot makes the code a more concise and allows more code reuse.

Reply:
Fair enough, I tend to want to separate the determination of a state from the decision. If that doesn't work in this case, that's ok.